app/controllers/recipes_controller.rb

```diff
@@ -68,6 +62,43 @@ def category_options
     options
   end
 
+  def ingredient_options
+    options = [[nil]]
+    Ingredient.all.each do |ingredient|
+      options << [ingredient.for_dropdown, ingredient.id]
+    end
+    options
+  end
+
+  def add_recipe_categories
+    recipe_category_ids.each do |key, category_id|
+      RecipeCategory.create(
+        recipe: @recipe,
+        category_id: category_id[:id])
+    end
+  end
+
+  def add_ingredient_lists
+    ingredient_lists.each do |key, info|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.